src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -2413,16 +2429,18 @@ MulticopterPositionControl::calculate_velocity_setpoint()
 	}
 
 	// encourage pilot to respect flow sensor minimum height limitations
-	if (_local_pos.limit_hagl && _local_pos.dist_bottom_valid && _control_mode.flag_control_manual_enabled
+	/*
```

Comment:
Why has this been removed? This functionality is required for reliable flow operation on some platforms.

Reply:
I removed this since it had been preventing proper landing in POSCTL. The vehicle would keep trying to climb when commanded to land, and the land detector was never triggering. This led to the vehicle flipping on the ground. 

This solution also seems a bit hacky. @Stifael any suggestions here?

Reply:
During my testing I had enough vertical sink rate authority to overcome the small height rate bias and make it land.

We could use full down stick or down stick below 50% as a condition that turns of the effect.

Reply:
The method sounds good, but we need to think about this to make sure it's not a bunch of intransparent if statements.

@dagar suggestions?

Reply:
> I removed this since it had been preventing proper landing in POSCTL.

 That is also what I realized. In fact, if you choose the minim altitude quite large, depending on the setup the vehicle will not be able to descend properly.

Please note that I started to port these kind of things into FlightTask (https://github.com/PX4/Firmware/pull/9368). I made a couple changes to the flow / terrain following logic.

For the bug above, the way I solved it was by only applying the limitation when no user altitude change is required (https://github.com/PX4/Firmware/blob/7952da65df77d78bad20c976cdd274b3adacd2de/src/lib/FlightTasks/tasks/FlightTaskManualAltitude.cpp#L109-L117).